Pax terminal net section (#1757)
SwedbankPay/developer.swedbankpay.com: modules-sdks/pax-terminal/instore-solution/NET/index.md

---
section: NET
title: Introduction and Overview
permalink: /:path/
redirect_from: /modules-sdks/pax-terminal/instore-solution/NET
description: |
    Use the .NET SDK to quickly and easy integrate with the terminal from your POS solution. The aim of the SDK is to minimize the work effort for both users and Swedbank Pay.
menu_order: 1500
---

## Introduction

SwpTrmLib is a Net Standard 2.0 nuget package that may be used when integrating Swedbank Pay terminal to a sale system. The
package makes it simple and easy to get started.
It has high abstraction and even if the actual protocol to the terminal changes this interface can stay the same.

It offers asynchronous methods and for the [simplest form of implementation][client-style] it just takes a few calls to make a transaction.

As of now the terminal is not prepared for a cloud connection, but by using this SDK it is possible to quickly make a small proxy service that makes whichever cloud connection that is desired.

> In the future there may be other variants of the interface and usage within the SDK, but this version will stay the same and once an implementation has been made it should not have to be changed unless new requirements or new functionallity is desired.

Reviewer: ```suggestion
In the future there may be other variants of the interface and usage within the SDK, but this version will stay the same and once an implementation has been made, it should not have to be changed unless new requirements or new functionality is desired.
```

* [Create][create-method] - Creates an instance and returns an interface
* [Start][start-method] - Initialises the instance and starts a listener for terminal if that mode is chosen
* [OpenAsync][openasync] - Starts a Login Session with the terminal. The session remains until Close or a new Open call
* [PaymentAsync][paymentasync] - Starts a payment transaction for supplied amount.
* [RefundAsync][refundasync] - Starts a refund transaction for supplied amount.
* [CloseAsync][closeasync] - Finnish the terminal session and allows for terminal maintenance. At least once a day

## Handy Methods

To get more than just payments and refunds

* [GetPaymentInstrumentAsync][getpaymentinstrumentasync] - Opens card readers to read card befor amount is known
* [SetPaymentInstrument][setpaymentinstrument] - Send a payment instrument (card number) to the terminal. Note! Only non PCI regulated cards.
* [RequestToDisplayAsync][requesttodisplayasync] - Send message to be displayed on terminal
* [RequestCustomerConfirmationAsync][requestcustomerconfirmation] - Display a yes/no question on the terminal and receive the result.

## Other Available Methods

> * [AbortAsync][abortasync] - Abort something ongoing

Reviewer: ```suggestion
* [AbortAsync][abortasync] - Aborts something ongoing
```

> * [ReverseLastAsync][reverselastasync] - Reverse last transaction if it was approved

Reviewer: ```suggestion
* [ReverseLastAsync][reverselastasync] - Reverses the last transaction if it was approved
```
